[Term Entry]  Javascript Method :  .toLocaleString()

lamiaelhbari opened the pull request:
<!--
👋 Hi, thanks for submitting a PR to Codecademy Docs! 💖
Please fill out all fields below to ensure your PR is reviewed quickly.

**IMPORTANT**

If you would like to receive credit for your contribution to an entry, make sure to have your Codecademy user profile linked to your GitHub account:

1. Go to your Codecademy dashboard.
2. Click on your profile image in the top-right and then choose "Profile".
3. Then click "Edit Profile".
4. In the "GitHub Username" field, add your username (without the @).
5. Click "Save Changes"

Of course, you can opt not to do this and be listed as an "Anonymous contributor", instead. :)
-->

### Description

<!-- Please write a summary of the change, such as which topic(s) and file(s) that you have edited or created. Please also include relevant motivation and context: -->
- Added the .toLocaleString() term for javascript 
- Closes Issue #3997 

### Issue Solved

<!--
Please use the following format(s) to link the issue numbers that your contribution addresses:
Closes #issue-number (where `issue-number` corresponds to the PRs' respective issue)
-->

### Type of Change

<!--- Please delete or cross off the bullet point(s) that are irrelevant to this PR: -->

- Adding a new entry 

### Checklist

<!-- Please check ALL the boxes: -->

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

<!--
Having trouble with the PR checker? Here are some common issues and resolutions:

- verify_formatting is failing
  - run `yarn format path/to/markdown/file.md` or `yarn format:all` and commit the results
- verify_lint is failing
  - same as above
  - if verify_lint is still failing, running `yarn lint` locally should let you know what needs to be changed by hand
- test is failing
  - ensure any new markdown files have a `Title` and `Description` defined in their metadata
  - ensure any new markdown files only contain alphanumerics and dashes in their file names and have the same name as their parent directory
  - if that looks ok, running `yarn test` locally should let you know what the issue is
-->

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/Codecademy/docs?pullRequest=4608) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/Codecademy/docs?pullRequest=4608) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Codecademy/docs?pullRequest=4608) it.</sub>

mamtawardhani commented on content/javascript/concepts/dates/terms/toLocaleString/toLocaleString.md:
````diff
@@ -0,0 +1,89 @@
+```js {copy}
+toLocaleString(locales [, options])
+```
+- `locales` : Represents the `locale or locales` to use when formatting the `date`.
+- `options` : Contains additional formatting options for the `date`, such as the format of the `weekday`, `month`, `year`, etc.
+
````
```suggestion

- `dateObj`: The `Date` object used to format as a string based on the locale.
- `locales`: A `string` or an `array of strings` that specifies one or more `locales` or language tags for formatting the date. 
- `options`: An object that allows customizing the formatting behavior, such as specifying the format for `date`, `time`, `numeric values`, and more.

> **Note:** Both the `locales` and `options` parameters in the `.toLocaleString()` method are optional. If the parameters are not provided, the method will use `default` values based on the runtime environment.
 ```

mamtawardhani commented on content/javascript/concepts/dates/terms/toLocaleString/toLocaleString.md:
```diff
@@ -0,0 +1,89 @@
+
+The `.toLocaleString()` method in JavaScript is used to format a [`Date` object](https://www.codecademy.com/resources/docs/javascript/dates), as a string based on the specified locale. This means it takes into account cultural settings such as `language` and `date/time` formatting preferences specific to the selected region or country.
+
+## Syntax : 
```
```suggestion
## Syntax
```

mamtawardhani commented on content/javascript/concepts/dates/terms/toLocaleString/toLocaleString.md:
```diff
@@ -0,0 +1,89 @@
+- `options` : Contains additional formatting options for the `date`, such as the format of the `weekday`, `month`, `year`, etc.
+
+
+## Example : 
```
```suggestion
## Example
```

mamtawardhani commented on content/javascript/concepts/dates/terms/toLocaleString/toLocaleString.md:
````diff
@@ -0,0 +1,89 @@
+
+## Example : 
+
+```js {copy}
+const date = new Date(); // Current date and time
+const options = { weekday: 'long', year: 'numeric', month: 'long', day: 'numeric' };
+console.log(date.toLocaleString('en-US', options));
+```
````
Please modify this example, the code has been plagiarized.
Plagiarised from: `https://debbie.codes/blog/date-formatting/`

mamtawardhani commented on content/javascript/concepts/dates/terms/toLocaleString/toLocaleString.md:
````diff
@@ -0,0 +1,89 @@
+```
+In this example, `toLocaleString()` formats the current `date` and `time` according to the long date format with the full `weekday`, `month`, `day`, and `year` in English (United States) `locale`.
+The output could be something like: 
+```shell {copy}
+// Output : "Sunday, May 5, 2024"
+```
````
Please modify the output and its description based on the new code example that you will use.

mamtawardhani commented on content/javascript/concepts/dates/terms/toLocaleString/toLocaleString.md:
````diff
@@ -0,0 +1,89 @@
+```
+Depending on the current `date` and `time` when the code is executed.
+
+
+## Runnable Code Example : 
+```js {copy}
+const date = new Date(1985,7,4);  
+const options = { weekday: 'long', year: 'numeric', month: 'long', day: 'numeric' };
+
+// Formatting for English (United States) :
+console.log(date.toLocaleString('en-US', options));
+
+// Formatting for Korean (South Korea) :
+console.log(date.toLocaleString('ko-KR', options));
+
+// Formatting for Hindi (India) : 
+console.log(date.toLocaleString('hi-IN', options));
+```
+In this example, we use `toLocaleString()` to format the current `date` and `time` with the specified formatting options for `English` (United States), `Korean` (South Korea), and `Hindi` (India).
+
+
+```shell {copy}
+// Output : 
+For English (United States): "Sunday, August 4, 1985" .
+For Korean (South Korea):  "1985년 8월 4일 일요일" .
+For Hindi (India): "रविवार, 4 अगस्त 1985" .
+```
+
+
````
The lines 45 to 70 can be removed. Only one ## Example and one ## Codebyte Example is needed.

mamtawardhani commented on content/javascript/concepts/dates/terms/toLocaleString/toLocaleString.md:
````diff
@@ -0,0 +1,89 @@
+
+In the following example the variables `locale` and `options` can be modified, to print the `Date` in a custom format:
+
+```codebyte/js
````
```suggestion
```codebyte/javascript
```

mamtawardhani commented on content/javascript/concepts/dates/terms/toLocaleString/toLocaleString.md:
````diff
@@ -0,0 +1,89 @@
+
+console.log(date.toLocaleDateString(locale, options));
+```
+
+
````
```suggestion
```

mamtawardhani commented on content/javascript/concepts/dates/terms/toLocaleString/toLocaleString.md:
````diff
@@ -0,0 +1,89 @@
+```shell {copy}
+// Output : "Sunday, May 5, 2024"
+```
+Depending on the current `date` and `time` when the code is executed.
+
````
```suggestion
```

mamtawardhani commented on content/javascript/concepts/dates/terms/toLocaleString/toLocaleString.md:
```diff
@@ -0,0 +1,89 @@
+  weekday: 'long',
+};
+
+console.log(date.toLocaleDateString(locale, options));
```
```suggestion
console.log(date.toLocaleString(locale, options));
```

mamtawardhani reviewed (changes requested):
Hey @lamiaelhbari, thank you for contributing to Codecademy Docs! 
The entry is well written, however, a few things are to be looked at:

(1) Please sign the CLA.
(2) I've provided suggestions and comments on various parts of this entry, please review and modify them at your earliest convenience.

Thank you!

lamiaelhbari commented:
Hi **@mamtawardhani** , Thank you for your detailed suggestions!

I will take your comments into consideration and make the necessary changes as soon as possible.
I will keep you informed of the progress. If you have any  questions or requests, please feel free to let me know. 

Thanks again for your assistance!

mamtawardhani commented:
Hello @lamiaelhbari! Could you please make the required changes at your earliest convenience?
Thanks!

lamiaelhbari commented:
Hello **@mamatawardhani,**

I have addressed all the requested changes and updated the branch. All CI checks have passed successfully. Could you please review the updates and approve the workflow?

Thank you for your time and assistance!

mamtawardhani reviewed (approved):
Thank you @lamiaelhbari for making the required changes!

Looks for the second review

avdhoottt commented on content/javascript/concepts/dates/terms/toLocaleString/toLocaleString.md:
```diff
@@ -0,0 +1,66 @@
+---
+Title: '.toLocaleString()'
+Description: 'Returns a string formatted based on the locale for a Date object in JavaScript.'
+Subjects:
+  - 'Computer Science'
+  - 'Web Development'
+Tags:
+  - 'Date'
+  - 'Methods'
+  - 'Strings'
+CatalogContent:
+   - 'introduction-to-javascript'
+   - 'paths/front-end-engineer-career-path'
```
```suggestion
Title: '.toLocaleString()'
Description: 'Returns a string formatted based on the locale for a Date object in JavaScript.'
Subjects:
  - 'Computer Science'
  - 'Web Development'
Tags:
  - 'Date'
  - 'Methods'
  - 'Strings'
CatalogContent:
  - 'introduction-to-javascript'
  - 'paths/front-end-engineer-career-path'
```

avdhoottt commented on content/javascript/concepts/dates/terms/toLocaleString/toLocaleString.md:
```diff
@@ -0,0 +1,66 @@
+   - 'paths/front-end-engineer-career-path'
+---
+
+In JavaScript, the `.toLocaleString()` method formats a [Date](https://www.codecademy.com/resources/docs/javascript/dates) object as a string according to the specified locale, considering cultural settings such as `language` and `date/time` formatting preferences specific to the chosen region or country.
```
```suggestion
In JavaScript, the `.toLocaleString()` method formats a [`Date`](https://www.codecademy.com/resources/docs/javascript/dates) object as a string according to the specified locale, considering cultural settings such as `language` and `date/time` formatting preferences specific to the chosen region or country.
```

avdhoottt commented on content/javascript/concepts/dates/terms/toLocaleString/toLocaleString.md:
````diff
@@ -0,0 +1,66 @@
+
+## Syntax  
+
+```js
````
```suggestion
```pseudo
```
The syntax should be inside a pseudo block.

avdhoottt commented on content/javascript/concepts/dates/terms/toLocaleString/toLocaleString.md:
````diff
@@ -0,0 +1,66 @@
+In the example below, `toLocaleString()` formats the current `date` and `time` according to the long date format with the full `weekday`, `month`, `day`, and `year` in English (United States) `locale`.
+
+```javascript {copy}
+const currentDate = new Date(); 
+const locale = 'en-US';
+const options = { weekday: 'long', year: 'numeric', month: 'long', day: 'numeric' };
+console.log(currentDate.toLocaleDateString(locale, options));
````
```suggestion
const specificDate = new Date('2024-05-16'); 
const locale = 'en-US';
const options = { weekday: 'long', year: 'numeric', month: 'long', day: 'numeric' };
console.log(specificDate.toLocaleDateString(locale, options));
```
The problem with this code is that the output would vary depending on the current date and the viewer will might get confused. Let's just specify the date so the output will be the same every time.

avdhoottt commented on content/javascript/concepts/dates/terms/toLocaleString/toLocaleString.md:
````diff
@@ -0,0 +1,66 @@
+
+In the example below, `toLocaleString()` formats the current `date` and `time` according to the long date format with the full `weekday`, `month`, `day`, and `year` in English (United States) `locale`.
+
+```javascript {copy}
````
```suggestion
```js
```

avdhoottt commented on content/javascript/concepts/dates/terms/toLocaleString/toLocaleString.md:
````diff
@@ -0,0 +1,66 @@
+The code above produces the following output:
+
+```shell
+Monday, May 6, 2024
````
```suggestion
Thursday, May 16, 2024
```

avdhoottt commented on content/javascript/concepts/dates/terms/toLocaleString/toLocaleString.md:
```diff
@@ -0,0 +1,66 @@
+
+## Codebyte Example
+
+In the following example, `.toLocaleString()` formats the current date and time with the full `weekday`, `year`, `month`, `day`, `hour`, `minute`, and `second` in French (Morocco) locale (fr-MA).
```
```suggestion
In the following example, `.toLocaleString()` formats the current date and time with the full `weekday`, `year`, `month`, `day`, `hour`, `minute`, and `second` in French (Morocco) locale (fr-MA):
```

avdhoottt commented on content/javascript/concepts/dates/terms/toLocaleString/toLocaleString.md:
```diff
@@ -0,0 +1,66 @@
+
+## Example  
+
+In the example below, `toLocaleString()` formats the current `date` and `time` according to the long date format with the full `weekday`, `month`, `day`, and `year` in English (United States) `locale`.
```
```suggestion
In the example below, `toLocaleString()` formats the current `date` and `time` according to the long date format with the full `weekday`, `month`, `day`, and `year` in English (United States) `locale`:
```

avdhoottt reviewed (changes requested):
Hey @lamiaelhbari, excellent entry! I have suggested some changes, please make them.

mamtawardhani commented:
Hey @lamiaelhbari are you still working on this?

avdhoottt commented on content/javascript/concepts/dates/terms/toLocaleString/toLocaleString.md:
```diff
@@ -0,0 +1,66 @@
+- `options`: An object that allows customizing the formatting behavior, such as specifying the format for `date`, `time`, `numeric values`, and more.
+
+> **Note:** Both the `locales` and `options` parameters in the `.toLocaleString()` method are optional. If the parameters are not provided, the method will use `default` values based on the runtime environment.
+
```
```suggestion
```

avdhoottt reviewed (approved):
Made the changes and merged this PR myself as the author is not responding.
